jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleLayerConfigurations.java

```diff
@@ -75,87 +71,76 @@ static JavaLayerConfigurations getForProject(
   private static JavaLayerConfigurations getForNonWarProject(
       Project project, Logger logger, Path extraDirectory, AbsoluteUnixPath appRoot)
       throws IOException {
+    Builder layerBuilder = JavaLayerConfigurations.builder();
+
     JavaPluginConvention javaPluginConvention =
         project.getConvention().getPlugin(JavaPluginConvention.class);
-
     SourceSet mainSourceSet = javaPluginConvention.getSourceSets().getByName(MAIN_SOURCE_SET_NAME);
 
-    List<Path> dependenciesFiles = new ArrayList<>();
-    List<Path> snapshotDependenciesFiles = new ArrayList<>();
-    List<Path> classesFiles = new ArrayList<>();
-    List<Path> extraFiles = new ArrayList<>();
+    FileCollection classesDirectories = mainSourceSet.getOutput().getClassesDirs();
+    Path resourcesOutputDirectory = mainSourceSet.getOutput().getResourcesDir().toPath();
+    FileCollection allFiles = mainSourceSet.getRuntimeClasspath();
+    FileCollection dependencyFiles =
+        allFiles
+            .minus(classesDirectories)
+            .filter(file -> !file.toPath().equals(resourcesOutputDirectory));
 
-    // Adds each file in each classes output directory to the classes files list.
-    FileCollection classesOutputDirectories = mainSourceSet.getOutput().getClassesDirs();
+    // Adds class files.
     logger.info("Adding corresponding output directories of source sets to image");
-    for (File classesOutputDirectory : classesOutputDirectories) {
-      if (Files.notExists(classesOutputDirectory.toPath())) {
-        logger.info("\t'" + classesOutputDirectory + "' (not found, skipped)");
-        continue;
-      }
-      logger.info("\t'" + classesOutputDirectory + "'");
-      try (Stream<Path> classFileStream = Files.list(classesOutputDirectory.toPath())) {
-        classFileStream.forEach(classesFiles::add);
-      }
-    }
-    if (classesFiles.isEmpty()) {
+    classesDirectories
```

Comment:
This may be a small personal preference, but I found this logic (L90-101) easier to read when it was all under the same loop, rather than filtering out/iterating over missing and existing files separately.

Reply:
I see. I will revert it to do the logging in the same loop.